test/api/api_test.cpp

```diff
@@ -316,3 +316,31 @@ TEST_F(ApiTest, MissingParam) {
     ASSERT_FALSE(prep->isSuccess());
     ASSERT_STREQ("Parameter val1 not found.", prep->getErrorMessage().c_str());
 }
+
+TEST_F(ApiTest, CloseDatabaseBeforeQueryResultAndConnection) {
```

Comment:
Can we cover more public interfaces of `QueryResult` and `Connection` in tests?

Reply:
Done